symengine/polynomial.cpp

```diff
@@ -413,11 +413,11 @@ RCP<const UnivariatePolynomial> UnivariatePolynomial::from_vec(const RCP<const S
     return make_rcp<const UnivariatePolynomial>(var, degree, std::move(dict));
 }
 
-RCP<const Basic> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
+RCP<const UnivariatePolynomial> UnivariatePolynomial::from_dict(const RCP<const Symbol> &var, map_int_Expr &&d)
 {
     auto iter = d.begin();
     while (iter != d.end()) {
-        if (Expression(0) == iter->second) {
+        if (Expression(0) == iter->second and Expression(0) != iter->first) {
```

Comment:
@isuruf I have a doubt. This change is required for [this](https://github.com/symengine/symengine/pull/896/files#diff-8b7281251a4cce49b1d83e34bf33e6fbR275) test case to work. Without this `z->is_zero()` returns `false`. But this change is not required for a [similar test case](https://github.com/symengine/symengine/blob/master/symengine/tests/basic/test_polynomial.cpp#L136) in `UnivariateIntPolynomial`. Why so?